src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -922,7 +923,7 @@ private function registerProfilerConfiguration(array $config, ContainerBuilder $
             $container->removeDefinition('console_profiler_listener');
         }
 
-        if (!class_exists(CommandDataCollector::class)) {
+        if (!$container->getParameter('kernel.debug') || !class_exists(CliRequest::class) || !class_exists(CommandDataCollector::class)) {
```

Comment:
why this check for kernel.debug ? If you enable the profiler in non-debug mode (for instance in tests), you will need this.

Reply:
Good question, i’ll digg tomorrow

Reply:
This line will also be reverted in #52274.

Reply:
Let's continue discussion (or stop it) on #52274  then